Gems/Atom/RPI/Code/Tests/Model/ModelTests.cpp

```diff
@@ -1031,42 +1035,76 @@ namespace UnitTest
     static constexpr inline auto minmaxElement = AZStd::minmax_element(begin(TwoSeparatedPlanesIndices), end(TwoSeparatedPlanesIndices));
     static_assert(*minmaxElement.second == (TwoSeparatedPlanesPositions.size() / 3) - 1);
 
-    template<class x> class TD;
     class TestMesh
     {
     public:
+        TestMesh() = default;
+
         TestMesh(const float* positions, size_t positionCount, const uint32_t* indices, size_t indicesCount)
         {
             AZ::RPI::ModelLodAssetCreator lodCreator;
+            Begin(lodCreator);
+            Add(lodCreator, positions, positionCount, /*positionOffset=*/0, indices, indicesCount, /*indexOffset=*/0);
+            End(lodCreator);
+        }
+
+        // initiate the asset lod creation process (note: End must be called after meshes have been added).
+        void Begin(AZ::RPI::ModelLodAssetCreator& lodCreator)
+        {
             lodCreator.Begin(AZ::Data::AssetId(AZ::Uuid::CreateRandom()));
+        }
 
+        // add a sub mesh and reuse existing position/index buffer (be very careful with the offsets used)
+        void Add(
+            AZ::RPI::ModelLodAssetCreator& lodCreator,
+            const float* positions,
+            size_t positionCount,
+            size_t positionOffset,
+            AZ::Data::Asset<AZ::RPI::BufferAsset> positionBuffer,
+            const uint32_t* indices,
+            size_t indexCount,
+            size_t indexOffset,
+            AZ::Data::Asset<AZ::RPI::BufferAsset> indexBuffer)
+        {
             lodCreator.BeginMesh();
-            lodCreator.SetMeshAabb(AZ::Aabb::CreateFromMinMax({-1.0f, -1.0f, -0.5f}, {1.0f, 1.0f, 0.5f}));
+            lodCreator.SetMeshAabb(AZ::Aabb::CreateFromMinMax({ -1.0f, -1.0f, -0.5f }, { 1.0f, 1.0f, 0.5f }));
```

Comment:
These tests don't reply on the AABB so don't go through this code path which is why the values aren't super important, I can update this but can you please remove request changes as otherwise I'll be blocked all tomorrow and am out on Friday to merge, I'd like to get this fix merged soon now.

Reply:
Looks like our model class does the aabb check (using the AABB from the ModelAsset) then calls LocalRayIntersectionAgainstModel. We should really just move that inside LocalRayIntersectionAgainstModel. No reason I can think of to skip the AABB check.

No need to fix it, I created a GHI to follow up on it https://github.com/o3de/o3de/issues/5523